src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/lecture-unit-management.component.ts

```diff
@@ -186,6 +187,17 @@ export class LectureUnitManagementComponent implements OnInit, OnDestroy {
         });
     }
 
+    viewButtonAvailable(lectureUnit: LectureUnit) {
+        const pdfPattern = /\.pdf$/i;
+        switch (lectureUnit?.type) {
+            case LectureUnitType.ATTACHMENT:
+                const attachmentUnit = <AttachmentUnit>lectureUnit;
+                return pdfPattern.test(attachmentUnit.attachment?.link!);
```

Comment:
If you're confident that `attachmentUnit.attachment` is not undefined, you should use

```suggestion
                return pdfPattern.test(attachmentUnit.attachment!.link!);
```

because otherwise, you're suggesting to typescript that the value is both potentially undefined (because of the `?`) and not undefined (because of the `!`)